docs/tutorials/03_shaders/viz_sdf_cylinder.py

```diff
@@ -0,0 +1,347 @@
+        return mat4(1);
+
+    return mat4(Rp);
+}
+
+
+// SDF for the cylinder
+float sdCylinder( vec3 p, float r, float h )
+{
+    vec2 d = abs(vec2(length(p.xz),p.y)) - vec2(r,h);
+    return min(max(d.x,d.y),0.0) + length(max(d,0.0));
+}
+
+// This is used on calculations for surface normals of the cylinder
+float map( in vec3 position )
+{
+    mat4 rot = rotationMatrix(normalize(directionVSOutput), normalize(vec3(0, 1, 0)));
+
+    // this allows us to accommodate more than one object in the world space
+    vec3 pos = (rot*vec4(position - centerWCVSOutput, 0.0)).xyz;
+
+    // distance to the cylinder
+    return sdCylinder(pos, radiusVSOutput, heightVSOutput/2);
+}
+  
+  
+// We need surface normals when doing lighting of the scene
+vec3 calculateNormal( in vec3 position )
+{
+    vec2 e = vec2(0.001, 0.0);
+    return normalize( vec3( map(position + e.xyy) - map(position - e.xyy),
+                            map(position + e.yxy) - map(position - e.yxy),
+                            map(position + e.yyx) - map(position - e.yyx)));
+}
+
+
+// Ray Marching
+float castRay( in vec3 ro, vec3 rd )
+{
+    float t = 0.0;
+    for(int i=0; i < 4000; i++){
+        vec3 position = ro + t * rd;
+        float  h = map(position);
+        t += h;
+           if ( t > 20.0 || h < 0.001) break;
+    }
+    return t;
+}
+'''
+
+sdf_cylinder_frag_impl = \
+'''
+
+vec3 point = vertexMCVSOutput.xyz;
+
+// ray origin
+vec4 ro = -MCVCMatrix[3] * MCVCMatrix;  // camera position in world space
+
+vec3 col = vertexColorVSOutput.rgb;
+
+// ray direction
+vec3 rd = normalize(point - ro.xyz);
+
+// light direction
+vec3 ld = normalize(ro.xyz - point);
+
+ro += vec4((point - ro.xyz),0.0);
+
+float t = castRay(ro.xyz, rd);
+
+if(t < 20.0)
+{
+    vec3 position = ro.xyz + t * rd;
+    vec3 norm = calculateNormal(position);
+    float lightAttenuation = dot(ld, norm);
+    
+    // calculate the diffuse factor and diffuse color
+    df = max(0, lightAttenuation);
+    diffuse = df * diffuseColor * lightColor0;
+        
+    // calculate the specular factor and specular color
+    sf = pow(df, specularPower);
+    specular = sf * specularColor * lightColor0;
+    
+    // Blinn-Phong illumination model
+    fragOutput0 = vec4(ambientColor + diffuse + specular, opacity);
+}
+else{
+    discard;
+}
+'''
+
+###############################################################################
+# Finally, we add shader code implementation to the box_actor. We use
+# shader_to_actor to apply our implementation to the shader creation process,
+# this function joins our code to the shader template that FURY has by default.
+
+shader_to_actor(box_actor, "vertex", impl_code=sdf_cylinder_vert_impl,
+                decl_code=sdf_cylinder_vert_dec)
+shader_to_actor(box_actor, "fragment", decl_code=sdf_cylinder_frag_dec)
+shader_to_actor(box_actor, "fragment", impl_code=sdf_cylinder_frag_impl,
+                block="light")
+
+box_actor.GetProperty().SetRepresentationToSurface()
```

Comment:
If you remove the wireframe-related portion of the tutorial, then this line won't be necessary.